include/gunrock/algorithms/bfs.hxx

```diff
@@ -122,9 +168,15 @@ struct enactor_t : gunrock::enactor_t<problem_t> {
     };
 
     // Execute advance operator on the provided lambda
-    operators::advance::execute<operators::load_balance_t::block_mapped>(
-        G, E, search, context);
-
+    // Track performance stats if performance evaluation is turned on
+    if (performance) {
+      operators::advance::execute<operators::load_balance_t::block_mapped>(
+          G, E, search_with_performance, context);
+      *search_depth = iteration;
+    } else {
+      operators::advance::execute<operators::load_balance_t::block_mapped>(
+          G, E, search, context);
+    }
```

Comment:
I prefer a similar solution for betweenness centrality, where you can have an if-else block, and only allow the output frontier for the case where you need it profiled (collecting metrics).